sdk/Managed/src/Microsoft.WindowsAzure.MobileServices/Table/Sync/MobileServiceSyncContext.cs

```diff
@@ -370,6 +372,17 @@ private Task ExecuteOperationAsync(MobileServiceTableOperation operation, JObjec
 
                 try
                 {
+                    if (MobileServiceTableOperationKind.Insert != operation.Kind)
+                    {
+                        // Refresh the item's version, in case it changed while we were awaiting the item lock
+                        var localStoreItem = await this.Store.LookupAsync(operation.TableName, operation.ItemId);
```

Comment:
Correct. To be clear, I don't think this change should be taken as-is, but rather is just to show what I'm talking about in the comments for the previous commit and for issue #615. In the comments for this change, I state that "it doesn't account for when the initial push generates a legitimate merge conflict on the back-end, and the resolution is to accept the server version," which is exactly what you're saying. I think there has to be more infrastructure to call into client code and ask for the intent.  In my own client code, I call back into an IConflictResolver and get back a tuple with a new type-specific item plus an enum value that is either KeepLocal, KeepServer or ManualMerge, and the code in this change is indeed valid for the KeepLocal case.  In my own code, I have to fudge this in client code, which rapidly gets super complicated, and in the KeepServer version, to fix up the operations queue, I have to go manually delete rows from the operations table, which seems sketchy.  Thoughts?